src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -493,411 +493,1004 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveOptimization)]
-        public static unsafe int IndexOfAny(ref char searchSpace, char value0, char value1, int length)
+        public static unsafe int IndexOfAny(ref char searchStart, char value0, char value1, int length)
         {
             Debug.Assert(length >= 0);
 
-            fixed (char* pChars = &searchSpace)
-            {
-                char* pCh = pChars;
-                char* pEndCh = pCh + length;
+            nuint offset = 0; // Use nuint for arithmetic to avoid unnecessary 64->32->64 truncations
```

Comment:
Why `nuint` here, when `IndexOf` uses `nint`? Should that be changed to `nuint` as well?

https://github.com/benaadams/runtime/blob/73e8efb9f68bcb0e1c9c1ecf0f937447129a5373/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs#L218-L219

And if we change it, can we remove the `nint` overloads of `LoadVector128`, `LoadVector256`, etc.?